### PR TITLE
[web-bluetooth] - Updated advertising event with new props, updated non-nulls, non-nullable characteristic service

### PR DESCRIPTION
**Pull request by DaBs**
This is cleaning up some of the early types for scan events. I've attempted to mirror [the spec](https://webbluetoothcg.github.io/web-bluetooth/#advertising-events) as close as possible.
In the same vein, mark service as non-optional on `BluetoothRemoteGATTCharacteristic`. 
Also adding myself as code owner. I have a vested interest in the space and probably will continue to keep an eye on it, both in terms of spec and in terms of live functionality in regards to products.

As an aside, I've been considering how we could more meaningfully test the API changes, e.g. by potentially spinning up Chrome with the `FakeBluetooth` implementation and run tests against it. The current "tests" are quite meaningless. Not sure if this is too intensive, however.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://webbluetoothcg.github.io/web-bluetooth/#advertising-events
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by elibarzilay** [comment]
(Sounds like you'd *want* to add some tests no?)

File: types/web-bluetooth/index.d.ts [hunk 1]
```diff
@@ -61,12 +62,14 @@ interface BluetoothLEScan extends RequestLEScanOptions {
 }
 
 interface BluetoothAdvertisementEvent extends Event {
```
Comment (thegecko) [inline types/web-bluetooth/index.d.ts:64]: This is called `BluetoothAdvertisingEvent`

```suggestion
interface BluetoothAdvertisingEvent extends Event {
```

Comment (thegecko) [reply types/web-bluetooth/index.d.ts:64]: As this PR introduces breaking changes, suggest this naming is also fixed

**Review by thegecko — CHANGES_REQUESTED** [review]
Changes align with the spec, I've also suggested a further fix. Please do add tests, though

File: types/web-bluetooth/index.d.ts (hunk 1 above)
Comment (DaBs) [reply types/web-bluetooth/index.d.ts:64]: Agree, was considering it, we may as well align naming here too. I'll get it done.

**Comment by DaBs** [comment]
@thegecko Got a couple more changes when I started to really go through the spec. Also added initial type tests in accordance with the README

**Review by RyanCavanaugh — APPROVED** [review]

**Comment by DaBs** [comment]
Ready to merge

**Pull request merged**